### PR TITLE
DEV-149 | make sure https://github.com/heroku/php-getting-started should work locally

### PR DESCRIPTION
**Pull request by phuonglm**
Details: https://issues.teracy.org/browse/DEV-149

### CODE REVIEW

File: main-cookbooks/teracy-dev/recipes/apache.rb [hunk 1]
```diff
@@ -12,10 +12,41 @@
     node.default['apache']['listen_ports'] = [node['teracy-dev']['apache']['listen_port']]
     node.default['apache']['user'] = 'vagrant'
     node.default['apache']['group'] = 'vagrant'
+    node.default['apache']['version'] = node['teracy-dev']['apache']['version']
 
     apache_installed = Mixlib::ShellOut.new('which apache2').run_command.stdout
+    apache_version =  Mixlib::ShellOut.new('apache2 -v | grep -o "[0-9]\.[0-9]"').run_command.stdout
+    if apache_installed == '' or apache_version != node['teracy-dev']['apache']['version']
+        bash 'clean up previous install apapche' do
```
Comment (hoatle) [inline main-cookbooks/teracy-dev/recipes/apache.rb]: typo: apapche

File: main-cookbooks/teracy-dev/recipes/apache.rb [hunk 2]
```diff
@@ -12,10 +12,41 @@
...
+                apt-get install -y python-software-properties
+                apt-get autoremove
+                rm -rf /etc/apache2 && rm -rf /usr/lib/apache2
+            EOF
+            user 'root'
+        end
+        if node['teracy-dev']['apache']['version'] == '2.2'
+            bash 'remove apache 2.4 ppa' do
+                code <<-EOF
+                    add-apt-repository -y --remove ppa:ondrej/apache2
```
Comment (hoatle) [inline main-cookbooks/teracy-dev/recipes/apache.rb]: use apt_repository instead?

File: main-cookbooks/teracy-dev/recipes/apache.rb [hunk 3]
```diff
@@ -12,10 +12,41 @@
...
+                    apt-get update
+                EOF
+                user 'root'
+            end
+        else
+            bash 'add apache 2.4 ppa' do
+                code <<-EOF
+                    add-apt-repository -y ppa:ondrej/apache2
```
Comment (hoatle) [inline main-cookbooks/teracy-dev/recipes/apache.rb]: use apt_repository instead?

Comment (phuonglm) [inline main-cookbooks/teracy-dev/recipes/apache.rb]: Not tested yet but that possible. Not sure about any side effect compare to command line. But i feel more trust the bash command.

File: main-cookbooks/teracy-dev/recipes/apache.rb [hunk 4]
```diff
@@ -12,10 +12,41 @@
...
+                    apt-get update
+                EOF
+                user 'root'
+            end
+        end
+        Mixlib::ShellOut.new('apt-get update').run_command
```
Comment (phuonglm) [inline main-cookbooks/teracy-dev/recipes/apache.rb]: hmm, that apt-get update need to be remove.

**Comment by hoatle** [comment]
@phuonglm pls update the PR, I will continue checking this when the internet speed is back to normal.

**Comment by hoatle** [comment]
```
$ apache2 -V
[Fri Jan 23 04:49:49.018579 2015] [core:warn] [pid 21634] AH00111: Config variable ${APACHE_LOG_DIR} is not defined
(2)No such file or directory: AH02297: Cannot access directory '/etc/apache2/${APACHE_LOG_DIR}/' for log file '${APACHE_LOG_DIR}/other_vhosts_access.log' defined at /etc/apache2/conf-enabled/other-vhosts-access-log.conf:2
AH00014: Configuration check failed
vagrant@vagrant:~$ apache2 -V
[Fri Jan 23 04:50:36.715058 2015] [core:warn] [pid 21635] AH00111: Config variable ${APACHE_LOG_DIR} is not defined
(2)No such file or directory: AH02297: Cannot access directory '/etc/apache2/${APACHE_LOG_DIR}/' for log file '${APACHE_LOG_DIR}/other_vhosts_access.log' defined at /etc/apache2/conf-enabled/other-vhosts-access-log.conf:2
AH00014: Configuration check failed
```

**Comment by hoatle** [comment]
Currently on develop branch:
```
$ apache2 -V
Server version: Apache/2.2.22 (Ubuntu)
Server built:   Jul 22 2014 14:35:32
Server's Module Magic Number: 20051115:30
Server loaded:  APR 1.4.6, APR-Util 1.3.12
Compiled using: APR 1.4.6, APR-Util 1.3.12
Architecture:   64-bit
Server MPM:     Worker
  threaded:     yes (fixed thread count)
    forked:     yes (variable process count)
Server compiled with....
 -D APACHE_MPM_DIR="server/mpm/worker"
 -D APR_HAS_SENDFILE
 -D APR_HAS_MMAP
 -D APR_HAVE_IPV6 (IPv4-mapped addresses enabled)
 -D APR_USE_SYSVSEM_SERIALIZE
 -D APR_USE_PTHREAD_SERIALIZE
 -D SINGLE_LISTEN_UNSERIALIZED_ACCEPT
 -D APR_HAS_OTHER_CHILD
 -D AP_HAVE_RELIABLE_PIPED_LOGS
 -D DYNAMIC_MODULE_LIMIT=128
 -D HTTPD_ROOT="/etc/apache2"
 -D SUEXEC_BIN="/usr/lib/apache2/suexec"
 -D DEFAULT_PIDLOG="/var/run/apache2.pid"
 -D DEFAULT_SCOREBOARD="logs/apache_runtime_status"
 -D DEFAULT_ERRORLOG="logs/error_log"
 -D AP_TYPES_CONFIG_FILE="mime.types"
 -D SERVER_CONFIG_FILE="apache2.conf"
```

**Comment by phuonglm** [comment]
APACHE_LOG_DIR Is build pack specify variable it not relate to foreman or Ubuntu OS.
```
export APACHE_LOG_DIR="/tmp/"
```

**Comment by hoatle** [comment]
don't need anymore.

**Pull request closed without merging**